### PR TITLE
Fix trailing slash support

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

File: src/Matcher.js [hunk 1]
```diff
@@ -125,10 +125,18 @@ export default class Matcher {
       params[name] = value && decodeURIComponent(value);
     });
 
-    return {
-      params,
-      remaining: pathname.slice(match[0].length),
-    };
+    let remaining = pathname.slice(match[0].length);
+    if (this.endsWithDelimiter(match[0])) {
```
Comment (taion) [inline src/Matcher.js]: This is sort of interesting and comes up for the "named child routes" bit. If you had a route definition like:

```js
<Route path="foo">
  <Route path="*" />
</Route>
```

Would you expect the child route to match on a path of `/foo`? What about on a path of `/foo/`?

For something like:

```js
<Route path="foo">
  <Route path=":param" />
</Route>
```

The answer will be "no" for both.

Comment (taion) [inline src/Matcher.js]: Concretely, how bad is it to write:

```js
<Route
  path="foo"
  groups={{
    nav: (
      <Route Component={FooNav}>
        <Route path="*" />
      </Route>
    ),
    // ...
  }}
/>
```

v.:

```js
<Route
  path="foo"
  groups={{
    nav: (
      <Route path="*" Component={FooNav} />
    ),
    // ...
  }}
/>
```

Comment (jquense) [inline src/Matcher.js]: what is the overall question here? It seems like it'd be really weird for the child to match on `/foo` no?

Comment (itajaja) [inline src/Matcher.js]: what does rr do? regardless of what decision you want to take, I do believe that /foo and /foo/ should def be equivalent

Comment (taion) [inline src/Matcher.js]: Well it'd make the syntax for the nav very slightly easier in the examples above.

**Comment by taion** [comment]
This might be silly. Why would people ever not want to be loose on trailing slash support for a front end?

**Comment by jquense** [comment]
I don't think so, idk tho

**Comment by MartinDawson** [comment]
@taion `pathname` accepts trailing slashes so:

`/projects/` and `/projects`. We then build a url off of this pathname, e.g: `${location.pathname}/users but the problem we are having is that we are getting double slashes on the end if the user typed in `/projects/` manually which is causing a 404.

Would it not make sense for pathname to treat just remove slashes from the end? If not then where would be a good place to add this logic globally in found so we don't have to transform the `pathname` in each place manually?

**Pull request closed without merging**